_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_vaapi.cpp

```diff
@@ -1747,6 +1783,14 @@ mfxStatus VAAPIEncoder::Execute(Task const & task, mfxHDLPair pair)
                                                                  m_vaContextEncode, m_rirId), MFX_ERR_DEVICE_FAILED);
     }
 
+    //Gpu priority
+    if(m_MaxContextPriority != VA_ATTRIB_NOT_SUPPORTED)
```

Comment:
I believe we always had VAAPI attributes parsing contained in Query functions... I would suggest to have it there (see my prev. comment) and here just check:
```
if(m_MaxContextPriority)
{
   ...
}
```

Reply:
Done.